src/test/java/au/org/democracydevelopers/raireservice/request/GetAssertionsRequestTests.java

```diff
@@ -0,0 +1,146 @@
+/*
+  Copyright 2024 Democracy Developers
+  The Raire Service is designed to connect colorado-rla and its associated database to
+  the raire assertion generation engine (https://github.com/DemocracyDevelopers/raire-java).
+  
+  This file is part of raire-service.
+  raire-service is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+  raire-service is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details.
+  You should have received a copy of the GNU Affero General Public License along with ConcreteSTV.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package au.org.democracydevelopers.raireservice.request;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import au.org.democracydevelopers.raireservice.persistence.repository.ContestRepository;
+import java.math.BigDecimal;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("test-containers")
+@SpringBootTest
+@AutoConfigureTestDatabase(replace = Replace.NONE)
+public class GetAssertionsRequestTests {
+
+  @Autowired
+  ContestRepository contestRepository;
+
+   // A valid request for an IRV contest is valid.
+  @Test
+  public void validRequestForIRVContestIsValid() {
+    GetAssertionsRequest validRequest = new GetAssertionsRequest("Byron", List.of("Alice"), BigDecimal.valueOf(0.03));
+    assertDoesNotThrow(() -> validRequest.Validate(contestRepository));
+  }
+
+  // A request for a contest that doesn't exist is invalid.
+  @Test
+  public void requestForNonexistentContestIsInvalid() {
+    GetAssertionsRequest invalidRequest
+        = new GetAssertionsRequest("NonExistentContest", List.of("Alice"), BigDecimal.valueOf(0.03));
+    Exception ex = assertThrows(RequestValidationException.class, () -> invalidRequest.Validate(contestRepository));
+    assertTrue(ex.getMessage().toLowerCase().contains("No such contest".toLowerCase()));
+  }
+
+  // A request for an existent plurality contest is invalid.
```

Comment:
Might be worth indicating why this is invalid with an extra sentence. eg. "(This is invalid as assertions are only generated/stored for IRV contests)."